### ISSUE TITLE
Multiple subdomains using the same locale → first configured subdomain gets used

### ISSUE THREAD

**Issue by Poitrin** [vercel/next.js#22795]
### What version of Next.js are you using?

10.0.7 (but bug became visible already in 10.0.5)

I also tested with `"next": "^10.0.8-canary.17"`

### What version of Node.js are you using?

12.20.0

### What browser are you using?

Safari 14, Chrome 89

### What operating system are you using?

macOS Big Sur

### How are you deploying your application?

Custom server on AWS Lambda (serverless-express)

### Describe the Bug

1. Configure multiple domains (e.g. `qa.next.de`, `local.next.de`) that use the same `defaultLocale` (`de`)
2. Configure the hosts in `/etc/hosts` with all `*.next.*` domains pointing to `127.0.0.1` (see [Poitrin/nextjs-subdomains-test/README.md](https://github.com/Poitrin/nextjs-subdomains-test/blob/main/README.md))
3. Open `http://local.next.de:3000` in your Browser, with an index page that has `<Link href={'/subpage'}>...</Link>`
4. The `href` attribute of `next/link`'s `<Link>` component is `https://qa.next.de/subpage` on the server, and the client will complain…

```
Warning: Prop `href` did not match. Server: "https://qa.next.de/subpage" Client: "/subpage"
```

Bug appears both with `next dev` and on production.

### Expected Behavior

The link's `href` attributes on the server should not differ from the `href` attributes on the client (it should be `/subpage` in this case).

### To Reproduce

I created a minimal working example here:
https://github.com/Poitrin/nextjs-subdomains-test

**Comment by Poitrin** [comment]
Could get fixed with open PR #22337

**Comment by Poitrin** [comment]
See comment in now closed PR #22337:

> Hi, having multiple domain items with the same `defaultLocale` like this isn't currently supported and it looks like we are missing a check explaining this which I added [here](https://github.com/vercel/next.js/pull/27451) we already have an existing check [here](https://github.com/vercel/next.js/blob/9cbc4aacd50d0e79be7b8178eec9219457514da9/packages/next/server/config.ts#L353) to prevent accidentally defining locales for multiple domains.
> 
> This isn't supported as we currently don't know which domain to use when redirecting and creating links. I'm gonna close this as it seems like this should be a feature request/needs discussion instead of a bug fix.

**Comment by balazsorban44** [comment]
This issue has been automatically locked due to no recent activity. If you are running into a similar issue, please create a new issue with the steps to reproduce. Thank you.
